.github/workflows/test.yml

```diff
@@ -20,3 +20,12 @@ jobs:
       - uses: actions/checkout@v3
       - uses: nixbuild/nix-quick-install-action@v17
       - run: nix-build default.nix -A tests --arg supportedSystems '[ builtins.currentSystem ]' --restrict-eval -I . --allowed-uris 'https://github.com/NixOS/nixpkgs https://github.com/input-output-hk https://github.com/NixOS/nixpkgs-channels' --option trusted-public-keys "hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ= iohk.cachix.org-1:DpRUyj7h7V830dp/i6Nti+NEO2/nhblbov/8MW7Rqoo= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=" --option substituters "https://hydra.iohk.io https://iohk.cachix.org https://cache.nixos.org/"
+  pre-commit-check:
```

Comment:
This would replace the `nix-tests` action above

Reply:
So, `nix build .#pre-commit-check` implies `pre-commit run --all-files`, no way around it. 
It respects .gitignore, but it checks all files regardless of whether they've been modified by this commit. 
It's disadvantage is that it's slightly inefficient, but still lightning fast in practice.
It's advantage is that it will catch weird edge cases like it did in https://github.com/input-output-hk/plutus/pull/4886

The alternative is to replace with 
`nix develop --command pre-commit`
Which will work as expected. 

Which to pick?